Nested path support in hasProperty

Hi,

Great project; I am really finding this very useful. Thank you for your work.

I would like to upstream an enhancement to `hasProperty` which enables lodash `get/set` style property paths. Here is my current implementation.  

```javascript
const hasDeepProperty = _.curry((propertyOrPath, matcherOrValue) => {
  const pathParts = _.isArray(propertyOrPath) ? propertyOrPath : propertyOrPath.split('.')
  const matcher = _.reduceRight(pathParts, (m, p) => hasProperty(p, m), matcherOrValue)

  return _.assign(matcher, {
    describeTo (description) {
      description
      .append(pathParts.join('.') + ' be ')
      .appendDescriptionOf(matcherOrValue)
    },
  })
})
```
It recursively creates a matcher and then uses that matcher to do all of the dirty work. The whole function is current so it can be used to make tests more readable (i.e. `hasClassName = hasDeepProperty('props.className')`

Would you consider adding this change to the project? If so, I can make a pull request.

Thanks again.

Usage

```javascript
const hasClassName = hasDeepProperty('props.className')

assertThat(
  { props: { className: 'other-class some-flag-class' } },
  hasClassName(allOf(
    containsString('expected-class'),
    containsString('some-flag-class')
  ))
)

AssertionError: 
Expected: props.className be (a string containing "expected-class" and a string containing "some-flag-class")
     but: props className a string containing "expected-class": was "other-class some-flag-class"
```

--- a/lib/matchers/IsObjectWithProperties.js
+++ b/lib/matchers/IsObjectWithProperties.js
@@ -73,10 +73,11 @@ IsObjectWithProperties.hasProperties = function (properties) {
 	return new IsObjectWithProperties(properties);
 };
 
-IsObjectWithProperties.hasProperty = function (name, valueOrMatcher) {
-	const properties = {};
-	properties[name] = _.isUndefined(valueOrMatcher) ? defined() : valueOrMatcher;
-	return new IsObjectWithProperties(properties);
+IsObjectWithProperties.hasProperty = function (propertyOrPath, valueOrMatcher) {
+	const propertyPath = _.isArray(propertyOrPath) ? propertyOrPath : propertyOrPath.split('.');
+	const propertyToNestedMatcher = (matcher, prop) => new IsObjectWithProperties({[prop]: matcher});
+	const initialMatcher = _.isUndefined(valueOrMatcher) ? defined() : valueOrMatcher;
+	return _.reduceRight(propertyPath, propertyToNestedMatcher, initialMatcher);
 };
 
 module.exports = IsObjectWithProperties;

